blocks/hero/hero.js

```diff
@@ -1,16 +1,45 @@
-import '@udex/web-components/dist/HeroBanner.js';
-import {
-  div, h1, span, p,
-} from '../../scripts/dom-builder.js';
+import '@udex/webcomponents/dist/HeroBanner.js';
+import '@udex/webcomponents/dist/Button.js';
+import '@udex/webcomponents/dist/Avatar.js';
+import { div, span, p } from '../../scripts/dom-builder.js';
 import { getMetadata } from '../../scripts/aem.js';
 import { formatDate } from '../../scripts/utils.js';
 
-/*
-  * Assumption: The last paragraph is the description if it exists and has text
-*/
-function getDescription(block) {
-  const lastP = block.querySelector('p:last-of-type');
-  return lastP ? lastP.textContent?.trim() : '';
+function calculateInitials(name) {
+  const nameParts = name.split(' ');
+  let initials = '';
+  nameParts.forEach((part) => {
+    initials += part.charAt(0).toUpperCase();
+  });
+  return initials;
+}
+
+function decorateMetaInfo() {
+  const infoBlockWrapper = div({ class: 'media-blend__info-block' });
+
+  if (getMetadata('author')) {
+    const avatar = document.createElement('udex-avatar');
+    avatar.setAttribute('size', 'XS');
+    avatar.setAttribute('initials', calculateInitials(getMetadata('author')));
+    avatar.setAttribute('color-scheme', 'Neutral');
+
+    const author = span({ class: 'media-blend__author' }, getMetadata('author'));
+    infoBlockWrapper.append(avatar, author);
+  }
+  const lastUpdate = getMetadata('modified-time')
+    ? getMetadata('modified-time')
+    : getMetadata('published-time');
+  infoBlockWrapper.append(
+    span({ class: 'media-blend__date' }, `Updated on ${formatDate(lastUpdate)}`),
+  );
+
+  if (getMetadata('twitter:data2')) {
```

Comment:
so we are good with `twitter:data2` as reading time ?

Reply:
We have no other options right now. Changing it to `readingTime` would be nicer but we will give up `twitter:data2` field with that.